analytic_hr_department_restriction/models/ir_rule.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2025 Tecnativa - Víctor Martínez
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, models, tools
+from odoo.osv import expression
+from odoo.tools import config
+
+
+class IrRule(models.Model):
+    _inherit = "ir.rule"
+
+    @api.model
+    @tools.conditional(
+        "xml" not in config["dev_mode"],
+        tools.ormcache(
+            "self.env.uid",
+            "self.env.su",
+            "model_name",
+            "mode",
+            "tuple(self._compute_domain_context_values())",
+        ),
+    )
+    def _compute_domain(self, model_name, mode="read"):
+        """Inject extra domain for restricting plans and analytical accounts when the
+        user has the group 'Analytic Accounting'.
+        """
+        res = super()._compute_domain(model_name, mode=mode)
+        user = self.env.user
+        group1 = "base.group_user"
+        group2 = "analytic.group_analytic_accounting"
+        group3 = "analytic_hr_department_restriction.group_analytic_accounting_without_department"  # noqa: E501
+        group4 = "account.group_account_invoice"
+        test_condition = not config["test_enable"] or self.env.context.get(
+            "test_analytic_hr_department_restriction"
+        )
+        if (
+            model_name in ("account.analytic.account", "account.analytic.plan")
+            and not self.env.su
+            and user.has_group(group1)
+            and not user.has_group(group4)
+            and test_condition
+        ):
+            department_f_name = (
+                "department_id"
+                if model_name == "account.analytic.account"
+                else "department_ids"
+            )
+            if user.has_group(group2):
+                extra_domain = [
+                    (department_f_name, "child_of", user.employee_ids.department_id.ids)
```

Comment:
`child_of` works properly with many2many?

Reply:
It doesn't work correctly, I'm going to follow the way of making the company_dependent field https://github.com/OCA/account-analytic/pull/778#discussion_r2076890490